libraries/eosiolib/core/eosio/base64.hpp

```diff
@@ -0,0 +1,303 @@
+    // is not required as per RFC 2045.
+    //
+    // All chunks except the last one produce three output bytes.
+    //
+    // The last chunk produces at least one and up to three bytes.
+    //
+       eosio::check(pos+1 < length_of_string, "wrong encoded string size");
+       size_t pos_of_char_1 = pos_of_char(encoded_string.at(pos+1) );
+
+    //
+    // Emit the first output byte that is produced in each chunk:
+    //
+       ret.push_back(static_cast<std::string::value_type>( ( (pos_of_char(encoded_string.at(pos+0)) ) << 2 ) + ( (pos_of_char_1 & 0x30 ) >> 4)));
+
+       if ( ( pos + 2 < length_of_string  )       &&  // Check for data that is not padded with equal signs (which is allowed by RFC 2045)
+              encoded_string.at(pos+2) != '='     &&
+              encoded_string.at(pos+2) != '.'         // accept URL-safe base 64 strings, too, so check for '.' also.
+          )
+       {
+       //
+       // Emit a chunk's second byte (which might not be produced in the last chunk).
+       //
+          unsigned int pos_of_char_2 = pos_of_char(encoded_string.at(pos+2) );
+          ret.push_back(static_cast<std::string::value_type>( (( pos_of_char_1 & 0x0f) << 4) + (( pos_of_char_2 & 0x3c) >> 2)));
+
+          if ( ( pos + 3 < length_of_string )     &&
+                 encoded_string.at(pos+3) != '='  &&
+                 encoded_string.at(pos+3) != '.'
+             )
+          {
+          //
+          // Emit a chunk's third byte (which might not be produced in the last chunk).
+          //
+             ret.push_back(static_cast<std::string::value_type>( ( (pos_of_char_2 & 0x03 ) << 6 ) + pos_of_char(encoded_string.at(pos+3))   ));
+          }
+       }
+
+       pos += 4;
+    }
+
+    return ret;
+}
+
+std::string base64_decode(std::string const& s, bool remove_linebreaks) {
+   return decode(s, remove_linebreaks);
+}
+
+std::string base64_encode(std::string const& s, bool url) {
+   return encode(s, url);
+}
+
+std::string base64_encode(std::string_view s, bool url) {
+   return encode(s, url);
+}
+
+std::string base64_decode(std::string_view s, bool remove_linebreaks) {
+   return decode(s, remove_linebreaks);
+}
+
+} // anonymous namespace
+
+
+std::string base64_encode(unsigned char const* bytes_to_encode, unsigned int in_len) {
+   return detail::base64_encode(bytes_to_encode, in_len, false);
+}
+std::string base64_encode(std::string_view enc) {
+   return detail::base64_encode(enc, false);
+}
+std::string base64_decode(std::string_view encoded_string) {
+   return detail::base64_decode(encoded_string, false);
+}
+
+std::string base64url_encode(unsigned char const* bytes_to_encode, unsigned int in_len) {
+   return detail::base64_encode(bytes_to_encode, in_len, true);
+}
+std::string base64url_encode(std::string_view enc) {
+   return detail::base64_encode(enc, true);
+}
+std::string base64url_decode(std::string_view encoded_string) {
+   return detail::base64_decode(encoded_string, true);
```

Comment:
honestly I have no idea why author of the code decided to do so. that seems to be old third party code that I copied from `leap`. It seems that if `\n` appears in url encode then it will be missing in url decode.
anyway for bls we are not using that part. So we can remove url encode at all. What do you think?